From 30f3b0678ce6fc46bcdde61ca35821032d0f25b5 Mon Sep 17 00:00:00 2001
From: Arnd Bergmann <arnd@arndb.de>
Date: Thu, 2 Nov 2017 23:39:54 +0100
Subject: soc: amlogic: gx pm domain: add PM and OF dependencies

The new driver introduces harmless warnings:

warning: (PM_RMOBILE && ARCH_RCAR_GEN1 && ARCH_RCAR_GEN2 && ARCH_R7S72100 && MESON_GX_PM_DOMAINS) selects PM_GENERIC_DOMAINS which has unmet direct dependencies (PM)
warning: (MESON_GX_PM_DOMAINS) selects PM_GENERIC_DOMAINS_OF which has unmet direct dependencies (PM_GENERIC_DOMAINS && OF)

This adds CONFIG_OF and CONFIG_PM dependencies to ensure it
will only be enabled in valid configurations.

Fixes: 75fcb5ca4b46 ("soc: amlogic: add Meson GX VPU Domains driver")
Signed-off-by: Arnd Bergmann <arnd@arndb.de>
---
 drivers/soc/amlogic/Kconfig | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/soc/amlogic/Kconfig b/drivers/soc/amlogic/Kconfig
index 1621bb2..b04f6e4 100644
--- a/drivers/soc/amlogic/Kconfig
+++ b/drivers/soc/amlogic/Kconfig
@@ -12,6 +12,7 @@ config MESON_GX_SOCINFO
 config MESON_GX_PM_DOMAINS
 	bool "Amlogic Meson GX Power Domains driver"
 	depends on ARCH_MESON || COMPILE_TEST
+	depends on PM && OF
 	default ARCH_MESON
 	select PM_GENERIC_DOMAINS
 	select PM_GENERIC_DOMAINS_OF
-- 
cgit v1.1

